src/libs/TransactionRecoveryApi/AxelarRecoveryApi.ts

```diff
@@ -60,40 +71,34 @@ export class AxelarRecoveryApi {
   }
 
   public async queryTransactionStatus(txHash: string): Promise<GMPStatusResponse> {
-    const res = await this.execGet(this.axelarCachingServiceUrl, {
+
+    const txDetails = await this.execGet(this.axelarCachingServiceUrl, {
       method: "searchGMP",
       txHash,
-    });
-    if (res[0] && res[0].is_not_enough_gas)
-      return {
-        status: GMPStatus.GAS_UNPAID,
-        details: (res && res[0] && res[0].error) || "no valid response",
-        call: res[0]?.call,
-      };
-    else if (!res || res?.length < 1 || (res[0] && res[0].status === GMPStatus.ERROR))
-      return {
-        status: GMPStatus.ERROR,
-        details: (res && res[0] && res[0].error) || "no valid response",
-        call: res[0]?.call,
-      };
-    else if (res && res[0] && res[0].status === GMPStatus.EXECUTED)
-      return {
-        status: GMPStatus.EXECUTED,
-        details: res[0].executed,
-        call: res[0]?.call,
-      };
-    else if (res && res[0] && res[0].status === GMPStatus.APPROVED) {
-      return {
-        status: GMPStatus.APPROVED,
-        details: "",
-        call: res[0]?.call,
-      };
-    } else
-      return {
-        status: GMPStatus.CALL,
-        details: "",
-        call: res[0]?.call,
-      };
+    }).then(res => res[0]);
+
+    if (!txDetails) return { status: "error_fetching_status" }
```

Comment:
should we consider the following cases to return  `error_fetching_status` as well?
- `res` is undefined
- `execGet` throw an error because api endpoint is down

Reply:
added this block which might capture it:

`    let isError = false;

    const txDetails = await this.execGet(this.axelarCachingServiceUrl, {
      method: "searchGMP",
      txHash,
    })
      .then((res) => res[0])
      .catch((e) => {
        isError = true;
      });

    if (!txDetails || isError) return { status: "error_fetching_status" };`